App/Domain/Members/StatusName.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Domain\Members;
+
+enum StatusName: int
+{
+    case PENDING = 0;
+    case ACTIVE = 1;
+    case DEACTIVATED = 2;
+    case DELETED = 3;
```

Comment:
Why do you use int-backed enum? Wouldn't be better to use string-backed enums?

Reply:
Storing statuses as integers would be more beneficial for queries when it comes to performance

Reply:
In general, I agree. BUT 😉 

- it decreases friendliness of raw SQL queries (where you can't concat enum's value). You need to know statuses' values or check it every time to make query.
- in similar way it's less readable when looking at queries' results or logs
- in `App\Domain\Members\Id` you use string identifiers which makes your point less relevant. You probably would use members' IDs for searching more frequently than status. So if you allow "less performant" identifiers, why bother with status which is probably much better for indexing (limited amount of values)?
- it requires additional code for remapping integers to descriptive labels

I don't have any data to share, but I really doubt this would bring as significant performance improvement as it should be to consider all mentioned above things less important. Personally I would go with string statuses because I have positive experience with it and it was never a problem in apps I was working with 🙂.

Reply:
Sorry I don't have much experience and argument, but I agree with @Wirone and this article.
https://stitcher.io/blog/php-enum-style-guide#5.-no-index-values

I think it will hard to maintain